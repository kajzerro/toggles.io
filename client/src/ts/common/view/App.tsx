import * as React from 'react';
import { Router, Route, Switch, Redirect } from 'react-router';
import { connect } from 'react-redux';
import { History } from 'history';
import AuthPage from '../../auth/view/AuthPage';
import { State } from '../index';
import * as authActions from '../../auth/auth-actions';
import * as commonActions from '../../common/common-actions';
import { LogOutState } from '../../auth/auth-reducers';
import DashboardPage from './DashboardPage';
import NotFoundPage from './NotFoundPage';
import TopBar from './components/TopBar';
import TransientErrors from './components/TransientErrors';

interface OwnAppProps { dependencies: { routerHistory: History }; }

interface DispatchAppProps { onLogout: (fromAllDevices: boolean) => void; onRemoveError: (uuid: string) => void; }

interface StateAppProps {
  authorized: boolean;
  username: string;
  errors: Array<{ uuid: string, message: string }>;
  isLogoutLoading: boolean;
}

export type AppProps = OwnAppProps & StateAppProps & DispatchAppProps;
export function App(props: AppProps) {
  return (
    <Router history={props.dependencies.routerHistory}>
      <div>
        <TopBar isLoggedIn={props.authorized} onLogout={props.onLogout} username={props.username}
                isLogoutLoading={props.isLogoutLoading} />
        <Switch>
          <Route exact path="/">
            {props.authorized
              ? <Redirect to="/dashboard/projects" />
              : <Redirect to="/auth/login" />}
          </Route>
          <Route path="/auth" component={AuthPage} />
          <Route path="/dashboard" component={DashboardPage} />
          <Route component={NotFoundPage} />
        </Switch>
        <TransientErrors errors={props.errors} onRemove={props.onRemoveError} />
      </div>
    </Router>
  );
}

export default connect<StateAppProps, DispatchAppProps, OwnAppProps, State>(
  ({ auth, common }) => ({
    authorized: auth.user !== null,
    username: auth.user === null ? null : auth.user.username,
    errors: common.transientErrors,
    isLogoutLoading: auth.logOutState === LogOutState.PENDING
  }),
  (dispatch, ownProps) => ({
    onLogout: (fromAll: boolean) => dispatch(authActions.logOut({ fromAll })),
    onRemoveError: (uuid: string) => dispatch(commonActions.removeTransientError({ uuid }))
  })
)(App);
