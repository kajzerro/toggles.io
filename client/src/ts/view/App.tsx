import * as React from 'react';
import * as Radium from 'radium';

import * as styles from './App.styles';

export const UnstyledApp = () => <div style={[styles.common]}>Hello <strong>toggle.io</strong>!</div>;

export const App = Radium(UnstyledApp);
