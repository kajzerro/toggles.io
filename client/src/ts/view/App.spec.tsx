import 'raf/polyfill';

import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import { App } from './App';

const { render } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  it('should render a welcoming div', () => {
    const rendered = render(<App />);
    expect(rendered).toMatchSnapshot();
  });
});
