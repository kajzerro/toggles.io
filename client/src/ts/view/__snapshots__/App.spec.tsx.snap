// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a welcoming div 1`] = `
<div
  data-radium="true"
  style="width:100%;margin-top:20px;text-align:center;font-size:24px"
>
  Hello 
  <strong>
    toggle.io
  </strong>
  !
</div>
`;
